pkg/registry/admissionregistration/validatingadmissionpolicybinding/authz.go

```diff
@@ -85,26 +85,60 @@ func (v *validatingAdmissionPolicyBindingStrategy) authorize(ctx context.Context
 		}
 	}
 
+	var attrs authorizer.AttributesRecord
 	paramRef := binding.Spec.ParamRef
 
-	// require that the user can read (verb "get") the referred resource.
-	attrs := authorizer.AttributesRecord{
-		User:            user,
-		Verb:            "get",
-		ResourceRequest: true,
-		Name:            paramRef.Name,
-		Namespace:       paramRef.Namespace,
-		APIGroup:        apiGroup,
-		APIVersion:      apiVersion,
-		Resource:        resource,
+	//!NOTE: This applies a more restrictive requirement when using
+	// cluster-scoped params. Requires LIST of the resource paramKind.
+	// Can relax if we have access to discovery/restmapper to decide if
+	// paramKind is ClusterScoped so we can disambiguate the two cases.
+	switch {
+	case paramRef == nil:
+		// No authz check necessary
+	case len(paramRef.Name) > 0 && len(paramRef.Namespace) > 0:
+		// Require "get" of a specific resource
+		attrs = authorizer.AttributesRecord{
+			User:            user,
+			Verb:            "get",
+			ResourceRequest: true,
+			Name:            paramRef.Name,
+			Namespace:       paramRef.Namespace,
+			APIGroup:        apiGroup,
+			APIVersion:      apiVersion,
+			Resource:        resource,
+		}
+	case len(paramRef.Namespace) > 0:
+		// Require "list" of resource in given namespace
+		attrs = authorizer.AttributesRecord{
+			User:            user,
+			Verb:            "list",
+			ResourceRequest: true,
+			Name:            "", // list requests don't have names
+			Namespace:       paramRef.Namespace,
+			APIGroup:        apiGroup,
+			APIVersion:      apiVersion,
+			Resource:        resource,
+		}
+	default:
+		// Require "list" of resource in all namespace
+		attrs = authorizer.AttributesRecord{
+			User:            user,
+			Verb:            "list",
+			ResourceRequest: true,
+			Name:            "", // list requests don't have names
+			Namespace:       "", // no namespace indicates list across all namespaces
+			APIGroup:        apiGroup,
+			APIVersion:      apiVersion,
+			Resource:        resource,
+		}
```

Comment:
these two cases can collapse to a single case (set paramRef.Namespace in the authz attrs)

Reply:
done